### DISCUSSION TITLE
Thumbnails

### DISCUSSION THREAD

**Discussion by spicy-sauce** [strapi/strapi#7874] [Legacy Topics]
Seems that when uploading a gif file, a thumbnail is not created while for other formats it is.
Is it by design? 

Thanks

**Comment by derrickmehaffy** [comment]
Hi @spicy-sauce can you clarify your Strapi version and if you are using a 3rd party provider?

**Comment by spicy-sauce** [comment]
Hi,
Strapi version: 3.0.5
Not using a 3rd party provider, but the out of the box Upload plugin.

Thanks!

**Reply by derrickmehaffy** [reply]
Can you please test on a fresh project using the latest release `v3.1.4`

**Comment by spicy-sauce** [comment]
To clarify, none of the format is created to an uploaded gif file, while for others I get, for example:
large	Object { ext: ".jpeg", url: "/uploads/large_Whats_App_Image_2019_11_08_at_16_21_50_07d315fbd3.jpeg", hash: "large_Whats_App_Image_2019_11_08_at_16_21_50_07d315fbd3", … }
small	Object { ext: ".jpeg", url: "/uploads/small_Whats_App_Image_2019_11_08_at_16_21_50_07d315fbd3.jpeg", hash: "small_Whats_App_Image_2019_11_08_at_16_21_50_07d315fbd3", … }
medium	Object { ext: ".jpeg", url: "/uploads/medium_Whats_App_Image_2019_11_08_at_16_21_50_07d315fbd3.jpeg", hash: "medium_Whats_App_Image_2019_11_08_at_16_21_50_07d315fbd3", … }
thumbnail	Object { ext: ".jpeg", url: "/uploads/thumbnail_Whats_App_Image_2019_11_08_at_16_21_50_07d315fbd3.jpeg", hash: "thumbnail_Whats_App_Image_2019_11_08_at_16_21_50_07d315fbd3", … }

**Comment by MattieBelt** [comment]
Hey @spicy-sauce, this looks like intended, check out the source-code I found:
https://github.com/strapi/strapi/blob/d7ba1c2a3b6303a42feed03c6b8a2f1de037aa4e/packages/strapi-plugin-upload/services/image-manipulation.js#L146

Strapi uses `sharp` to resize the images and `sharp` recently added support for `.gif`: [#2012](https://github.com/lovell/sharp/pull/2012), not sure if this will be possible soon, but the `sharp` dependency recently upgraded. #7729 .

**Reply by MattieBelt** [reply]
Do note the following: 
1) `sharp` doesn't support resize (yet) [#2275](https://github.com/lovell/sharp/issues/2275)
2) > Please keep in mind that to create a GIF file, libvips must be compiled with imageMagick support

**Comment by spicy-sauce** [comment]
Thank you for your prompt reply 👍
